hook-bootkit/main.go

```diff
@@ -144,16 +147,35 @@ func main() {
 
 	fmt.Printf("Pulling image [%s]", imageName)
 
-	out, err := cli.ImagePull(ctx, imageName, pullOpts)
+	// TODO: Ideally if this function becomes a loop that runs forever and keeps retrying
+	// anything that failed, this retry would not be needed. For now, this addresses the specific
+	// race condition case of when the linuxkit network or dns is in the process of, but not quite
+	// fully set up yet.
+
+	var out io.ReadCloser
+	imagePullOperation := func() error {
+		out, err = cli.ImagePull(ctx, imageName, pullOpts)
+		if err != nil {
+			fmt.Printf("Image pull failure %s, %v\n", imageName, err)
+		}
+		return err
+	}
+	err = backoff.Retry(imagePullOperation, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetryAttempts))
+
 	if err != nil {
 		panic(err)
 	}
 
 	_, err = io.Copy(os.Stdout, out)
```

Comment:
you have a couple of single line if...err != nil and a couple 2 line error checks like this. Might want to pick one or the other only.

Reply:
there will be some mixed single and double line if statements based on whether a variable being set up is going to be used later. Example below:
`f, err := os.Open(name)
if err != nil {
    return err
}
codeUsing(f)`

fixed any fixable ones for consistency